ProcessQueue/__init__.py

```diff
@@ -266,6 +266,302 @@ def process_failure(api_url, scan_report_id, headers):
+    field_name_to_id_map={}
+    for name in names_list:
+        mappings_matching_field_name = [mapping for mapping in existing_mappings if mapping[0] == name]
+        target_concept_ids = set([mapping[1] for mapping in mappings_matching_field_name])
+        target_field_id = set([mapping[2] for mapping in mappings_matching_field_name])
+        if len(target_concept_ids) == 1:
+            field_name_to_id_map[str(name)] = str(target_field_id.pop())
+
+    
+    #replace field_name_to_id_map with field name to concept id map
+    #field_name_to_concept_id_map = { element.key: field_id_to_concept_map[int(element.value)] for element in field_name_to_id_map }
+    
+    print("FIELD NAME TO ID MAP",field_name_to_id_map)
+    concepts_to_post=[]
+    concept_response_content=[]
+    print("NAME IDS",new_field_ids.keys())
+
+    for name,id in new_field_ids.items():
+        try:
+            link_id=field_name_to_id_map[name]
+            concept_id=field_id_to_concept_map[int(link_id)]
+            
+            print(f"Found field with id: {link_id} with exsting concept mapping: {concept_id} which matches new field id: {id}")
+            # Create ScanReportConcept entry for copying over the concept
+            concept_entry={
+                    "nlp_entity": None,
+                    "nlp_entity_type": None,
+                    "nlp_confidence": None,
+                    "nlp_vocabulary": None,
+                    "nlp_processed_string": None,
+                    "concept":concept_id,
+                    "object_id":id,
+                    "content_type": content_type,
+                }
+            concepts_to_post.append(concept_entry)    
+        except KeyError:
+            continue
+    if concepts_to_post:
+
+        paginated_concepts_to_post = paginate_chars(concepts_to_post,"")
+        concept_response = []
+        for concepts_to_post_item in paginated_concepts_to_post:  
+            get_concept_response = requests.post(
+                    url=api_url + "scanreportconcepts/",
+                    headers=headers,
+                    data=json.dumps(concepts_to_post_item),
+                )
+            print("CONCEPTS SAVE STATUS >>>", get_concept_response.status_code,
+                    get_concept_response.reason, flush=True)
+            concept_response.append(json.loads(get_concept_response.content.decode("utf-8"))) 
+        concept_content = flatten(concept_response)
+
+        concept_response_content += concept_content
+
+        print("POST concepts all finished", datetime.utcnow().strftime("%H:%M:%S.%fZ"))
+
+def get_existing_value_concepts(values,content_type,api_url,headers):
+
+    """
+    This expects a list of values which have been generated in a newly uploaded scanreport and
+    creates new concepts if any matching names are found with existing fields
+    """
+    # create list of names of newly generated values
+    names_list = [value['value'] for value in values]
```

Comment:
Rename these like in the field function case, and reposition these lines so the structure more similarly matches that of the fields function.